Data/src/main/java/org/tribuo/data/columnar/processors/response/FieldResponseProcessor.java

```diff
@@ -64,20 +64,23 @@
     public void postConfig() {
         if (fieldName != null && fieldNames != null) {
             throw new PropertyException(configName, "fieldName, FieldNames", "only one of fieldName or fieldNames can be populated");
-        } else if (fieldNames != null) {
-            if(defaultValue != null) {
-                defaultValues = defaultValues == null ? Collections.nCopies(fieldNames.size(), defaultValue) : defaultValues;
+        } else if (fieldNames != null) { // multiple fields
+            if (defaultValues != null && defaultValues.size() == fieldNames.size()) { // check first for multiple defaults
+                // this is the default case, nothing needs to be done
```

Comment:
Can we flip this around to get rid of the empty if branch? Maybe start with `defaultValues == null && defaultValue == null` to catch the else case, and then have the `defaultValue != null` one, then `defaultValues != null && defaultValues.size() != fieldNames.size()` to throw `PropertyException`.